tests/ci_visibility/test_coverage_per_suite.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os
+
+import mock
+import pytest
+
+import ddtrace
+from ddtrace.contrib.pytest.plugin import is_enabled
+from ddtrace.internal import compat
+from ddtrace.internal.ci_visibility import CIVisibility
+from ddtrace.internal.ci_visibility.constants import COVERAGE_TAG_NAME
+from tests.ci_visibility.test_encoder import _patch_dummy_writer
+from tests.utils import TracerTestCase
+from tests.utils import override_env
+
+
+class PytestTestCase(TracerTestCase):
+    @pytest.fixture(autouse=True)
+    def fixtures(self, testdir, monkeypatch, git_repo):
+        self.testdir = testdir
+        self.monkeypatch = monkeypatch
+        self.git_repo = git_repo
+
+    def inline_run(self, *args):
+        """Execute test script with test tracer."""
+
+        class CIVisibilityPlugin:
+            @staticmethod
+            def pytest_configure(config):
+                if is_enabled(config):
+                    with _patch_dummy_writer():
+                        assert CIVisibility.enabled
+                        CIVisibility.disable()
+                        CIVisibility.enable(tracer=self.tracer, config=ddtrace.config.pytest)
+
+        with override_env(dict(DD_API_KEY="foobar.baz")):
+            return self.testdir.inline_run(*args, plugins=[CIVisibilityPlugin()])
+
+    @pytest.mark.skipif(compat.PY2, reason="ddtrace does not support coverage on Python 2")
+    def test_pytest_will_report_coverage_by_suite(self):
+        self.testdir.makepyfile(
+            test_ret_false="""
```

Comment:
very small nit: since these aren't meant to be test files, I'd avoid calling them `test_`